lib/fabrik/DirectorInstance.js

```diff
@@ -73,9 +76,12 @@ class DirectorInstance extends BaseInstance {
       .try(() => {
         switch (operation.type) {
         case 'create':
-          return this
-            .createSecurityGroup(operation.space_guid)
-            .tap(() => operation.state === CONST.OPERATION.SUCCEEDED ? this.scheduleAutoUpdate() : {});
+          if (operation.context.platform === 'cloudfoundry') {
```

Comment:
Instead of hardcoding platform specific processing with directorInstance like below, i propose the following. 
We would have class called `BaseCloudManager` which defines a basic methods / hooks like `preInstanceProvision`, `postInstanceProvision`, `preInstanceDelete`, `postInstanceDelete` etc... and then we can specific implementations of this baseclass/interface for CF & K8s as `CfCloudManager` & `K8sCloudManager`, which will have these platform specific implementations baked into them. 
Now when DirectorInstance is getting instantiated, it would be injected with the cloud specific manager into it & the `DirectorInstance` just invokes methods like `cloudManager.postInstanceProvision` or `cloudManager.postInstanceDelete` etc.. & so on. 
This keeps our code segregated, pluggable and modular even in our current state & this approach should not consume too much effort as well as the existing hardcoded implementations goes into specific `CloudManager` classes.

Reply:
Hi @jagadish-kb 

With SF 2.0 design, security group and other environment specific things will be executed as part of different controllers. Hence, I think different cloud specific Manager may not be necessary. In that case, all this code will do is trigger scheduleAutoUpdate.

I kept the code schedule autoupdate inside the if clause as at this moment K8S will not productively consume it. May be, I'll add this comment in the code. With SF2.0 it will be taken care of.

Does this sound good ?

Reply:
If we are adding any checks like the ones that we have done in the above, am suggesting we refactor such code in to a separate file. We don't even have to define/implement all the above suggested methods, whatever is applicable forn now we just move it to this separate implementation class (plaformmanager subclass).
Also, whatever u refactor from this and put in this plaformmanager classes can directly then go into ur respective controllers when the SF 2.0 code comes in to place, we dont have refactor DirectorInstance code again later. The part of the code that is common (DirectorInstance) will still get executed under one controller and whatever is part of these platform specific cloud manager can go into individual controllers. 
Whatever ur doing here could be thrown away or if u write it with a bit more planning aid to our next steps. Also make the current code look clean till the 2.0 implementation comes into effect (which certainly can take about min 2 months ?).

Reply:
As discussed with @jagadish-kb , we are not immediately puttin ghte platform checks in this PR. Security groups can be better handled as part of SF2.0 and hence we handle them there. For now, as CF is the only consumer, we dont check for the platform now.